helpers/export.js

```diff
@@ -64,16 +64,104 @@ function copyUntouchedTable(inputPath, outputPath, tableName, callback) {
 }
 
 function exportTable(tableName, gtfs, outputPath, callback) {
-  const csv = processGtfsTable(tableName, gtfs);
-
+  const actualKeys = gtfs.getActualKeysForTable(tableName);
+  const firstRow = `${actualKeys.join(',')}`;
   const outputFullPath = `${outputPath + tableName}.txt`;
-  fs.writeFile(outputFullPath, csv, (error) => {
-    if (error) {
-      throw error;
+
+  fs.writeFile(outputFullPath, firstRow, (writeError) => {
+    if (writeError) {
+      throw writeError;
+    }
+
+    const indexKeys = gtfs._schema.indexKeysByTableName[tableName];
+
+    if (indexKeys.singleton) {
+      let item = gtfs.getIndexedTable(tableName);
+      if (!item) {
+        callback();
+        return;
+      }
+
+      if (gtfs._preExportItemFunction) {
+        item = gtfs._preExportItemFunction(item, tableName);
+      }
+
+      const formattedGtfsRowValues = getObjectValuesUsingKeyOrdering(item, actualKeys);
+      const row = Papa.unparse({
+        fields: actualKeys,
+        data: formattedGtfsRowValues,
+      },
+      {
+        header: false,
+      });
+
+      fs.appendFile(outputFullPath, `\r\n${row}`, callback);
+      return;
     }
 
-    infoLog(`[${getHHmmss()}] Table has been exported: ${tableName}`);
-    callback();
+    const deepness = gtfs._schema.deepnessByTableName[tableName];
+
+    let rows = [];
+
+    async.eachSeries(gtfs.getIndexedTable(tableName), async ([key, object]) => {
```

Comment:
Also, to avoid confusion, I think we should let a small comment on why there is an `async` there.

Reply:
Good point. I will add.